subsys/net/l2/offloaded_if/offloaded_if.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(offloaded_if, LOG_LEVEL_DBG);
+
+#include <zephyr/net/net_l2.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/offloaded_if.h>
+
+static inline int offloaded_if_enable(struct net_if *iface, bool state)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct offloaded_if_api *off_if = dev->api;
+
+	if (!off_if || !(off_if->enable)) {
+		return 0;
+	}
+
+	/* Reviewers: Would it be better to pass the iface itself instead? */
+	return off_if->enable(dev, state);
+}
+
+NET_L2_INIT(OFFLOADED_IF, NULL, NULL, offloaded_if_enable, NULL);
```

Comment:
I feel it would be better not to mix `if` and `l2` concepts this way. I'd suggest to:
1. Keep the convention that `net/l2` is a place where you define L2s. Hence this fine and related functions/constructs would be named `offloaded_l2`/`OFFLOADED_L2`.

Reply:
Yeah, I actually originally named the L2 declared here as you recommend ("OFFLOADED_L2"), but I switched it to "OFFLOADED_IF" for a number of reasons:

Firstly, I thought this better reflected the fact that the L2 is used by offloaded ifaces/net-devices (specifically, those declared using Z_NET_DEVICE_OFFLOAD_INIT), much in the same sense that the L2 named `ETHERNET` is used by net devices declared with `Z_ETH_NET_DEVICE_INIT`

Secondly, `OFFLOADED_L2` would be a break from Zephyr convention since all the other L2 names eliminate the `L2` suffix from their names (notice there actually is a declaration for `OFFLOADED_IF_L2` that redefines it to just `OFFLOADED_IF` in the current commit). By this logic, the **_most_** accurate name would probably just be `OFFLOADED` but I fear using that word in isolation. 

I do admit, though, naming it `OFFLOADED_IF` is a bit problematic too, since it it's really the net device, not necessarily the iface specifically that is offloaded in some regard. And then there is the fact that it seems a bit oxymoronic to use the phrase `IF` in the name of an L2, as you point out

So I agree, I'm not happy with `OFFLOADED_IF` as a name, but all the other options seem equally flawed. @rlubos does a better name occur to you at all?

Reply:
Yes, using the network interface could be a bit misleading. Perhaps `OFFLOADED_NETDEV` or even `OFFLOADED_NET` could be used here.

Reply:
I like `OFFLOADED_NETDEV`! I worry about `OFFLOADED_NET` since that could be misconstrued as specifically associated with `CONFIG_NET_OFFLOAD`, which it is not. I think `OFFLOADED_NETDEV` will be the least confusing option among all that have been suggested